clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -802,7 +802,8 @@ class KernelObjVisitor {
   template <typename... Handlers>
   void VisitArrayElements(FieldDecl *FD, QualType FieldTy,
                           Handlers &... handlers) {
-    const ConstantArrayType *CAT = cast<ConstantArrayType>(FieldTy);
+    const ConstantArrayType *CAT =
+        SemaRef.getASTContext().getAsConstantArrayType(FieldTy);
```

Comment:
This can return null, so we need an assert that it isn't null.  Probably the same below as well.

Reply:
Getting here is subject to passing the ConstantArray check in the field checker. So this will not return NULL.

Reply:
We still need asserts, otherwise the static analyzer will highlight this.